airbyte-webapp/src/components/ui/NextTable/NextTable.tsx

```diff
@@ -1,43 +1,87 @@
 import { ColumnDef, flexRender, useReactTable, getCoreRowModel } from "@tanstack/react-table";
+import { ColumnMeta } from "@tanstack/table-core";
 import classNames from "classnames";
-import { PropsWithChildren } from "react";
+import { CSSProperties, PropsWithChildren } from "react";
 
 import styles from "./NextTable.module.scss";
 
-export interface TableProps<TData> {
-  data: TData[];
-  columns: Array<ColumnDef<TData>>;
-  onClickRow?: (data: unknown) => void;
+type TData<T> = T & {
+  error?: unknown;
+};
+
+export interface TableProps<T> {
   className?: string;
+  columns: Array<ColumnDef<TData<T>>>;
+  data: Array<TData<T>>;
+  erroredRows?: boolean;
+  light?: boolean;
+  onClickRow?: (data: TData<T>) => void;
 }
 
-export const NextTable = <TData,>({ columns, data, onClickRow, className }: PropsWithChildren<TableProps<TData>>) => {
+const getStyleFromColumnMeta = <T,>(meta: ColumnMeta<T, unknown>): CSSProperties => {
+  const style: CSSProperties = {};
+  if (meta?.customWidth) {
```

Comment:
any reason why we introduced the style props?
We can pass the desired style via css in the meta